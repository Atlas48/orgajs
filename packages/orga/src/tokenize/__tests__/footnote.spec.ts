import tok from "./tok";

describe("tokenize footnote", () => {
  it("knows footnotes", () => {
    expect(tok("[fn:1] a footnote")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "[fn:1]",
          "label": "1",
          "type": "footnote.label",
        },
        Object {
          "_text": "a footnote",
          "type": "text.plain",
          "value": "a footnote",
        },
      ]
    `);
    expect(tok("[fn:word] a footnote")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "[fn:word]",
          "label": "word",
          "type": "footnote.label",
        },
        Object {
          "_text": "a footnote",
          "type": "text.plain",
          "value": "a footnote",
        },
      ]
    `);
    expect(tok("[fn:word_] a footnote")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "[fn:word_]",
          "label": "word_",
          "type": "footnote.label",
        },
        Object {
          "_text": "a footnote",
          "type": "text.plain",
          "value": "a footnote",
        },
      ]
    `);
    expect(tok("[fn:wor1d_] a footnote")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "[fn:wor1d_]",
          "label": "wor1d_",
          "type": "footnote.label",
        },
        Object {
          "_text": "a footnote",
          "type": "text.plain",
          "value": "a footnote",
        },
      ]
    `);
  });

  it("knows these are not footnotes", () => {
    expect(tok("[fn:1]: not a footnote")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "[fn:1]",
          "label": "1",
          "type": "footnote.reference",
        },
        Object {
          "_text": ": not a footnote",
          "type": "text.plain",
          "value": ": not a footnote",
        },
      ]
    `);
    expect(tok(" [fn:1] not a footnote")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "[fn:1]",
          "label": "1",
          "type": "footnote.reference",
        },
        Object {
          "_text": " not a footnote",
          "type": "text.plain",
          "value": " not a footnote",
        },
      ]
    `);
    expect(tok("[[fn:1] not a footnote")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "[",
          "type": "text.plain",
          "value": "[",
        },
        Object {
          "_text": "[fn:1]",
          "label": "1",
          "type": "footnote.reference",
        },
        Object {
          "_text": " not a footnote",
          "type": "text.plain",
          "value": " not a footnote",
        },
      ]
    `);
    expect(tok("\t[fn:1] not a footnote")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "[fn:1]",
          "label": "1",
          "type": "footnote.reference",
        },
        Object {
          "_text": " not a footnote",
          "type": "text.plain",
          "value": " not a footnote",
        },
      ]
    `);
  });
});
