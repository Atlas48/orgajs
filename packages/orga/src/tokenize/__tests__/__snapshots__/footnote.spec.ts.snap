// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize footnote knows footnotes 1`] = `
Array [
  Object {
    "label": "1",
    "position": Object {
      "end": Object {
        "column": 6,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "footnote.label",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 17,
        "line": 0,
      },
      "start": Object {
        "column": 7,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize footnote knows footnotes 2`] = `
Array [
  Object {
    "label": "word",
    "position": Object {
      "end": Object {
        "column": 9,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "footnote.label",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 20,
        "line": 0,
      },
      "start": Object {
        "column": 10,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize footnote knows footnotes 3`] = `
Array [
  Object {
    "label": "word_",
    "position": Object {
      "end": Object {
        "column": 10,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "footnote.label",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 21,
        "line": 0,
      },
      "start": Object {
        "column": 11,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize footnote knows footnotes 4`] = `
Array [
  Object {
    "label": "wor1d_",
    "position": Object {
      "end": Object {
        "column": 11,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "footnote.label",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 22,
        "line": 0,
      },
      "start": Object {
        "column": 12,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize footnote knows these are not footnotes 1`] = `
Array [
  Object {
    "label": "1",
    "position": Object {
      "end": Object {
        "column": 6,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.footnote",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 22,
        "line": 0,
      },
      "start": Object {
        "column": 6,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize footnote knows these are not footnotes 2`] = `
Array [
  Object {
    "label": "1",
    "position": Object {
      "end": Object {
        "column": 7,
        "line": 0,
      },
      "start": Object {
        "column": 1,
        "line": 0,
      },
    },
    "type": "text.footnote",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 22,
        "line": 0,
      },
      "start": Object {
        "column": 7,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize footnote knows these are not footnotes 3`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
  Object {
    "label": "1",
    "position": Object {
      "end": Object {
        "column": 7,
        "line": 0,
      },
      "start": Object {
        "column": 1,
        "line": 0,
      },
    },
    "type": "text.footnote",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 22,
        "line": 0,
      },
      "start": Object {
        "column": 7,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize footnote knows these are not footnotes 4`] = `
Array [
  Object {
    "label": "1",
    "position": Object {
      "end": Object {
        "column": 7,
        "line": 0,
      },
      "start": Object {
        "column": 1,
        "line": 0,
      },
    },
    "type": "text.footnote",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 22,
        "line": 0,
      },
      "start": Object {
        "column": 7,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;
